### PR TITLE
James 3797 Moving all source code related to SpamAssassin to module third-party/spamassassin

### PR DESCRIPTION
**Pull request by vttranlina**
https://issues.apache.org/jira/browse/JAMES-3797
https://github.com/linagora/james-project/issues/4588

### CODE REVIEW

File: server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java [hunk 1]
```diff
@@ -1,694 +0,0 @@
...
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
-
-import io.restassured.RestAssured;
-import io.restassured.builder.RequestSpecBuilder;
-import io.restassured.http.ContentType;
-import io.restassured.parsing.Parser;
-
-public interface SpamAssassinContract {
```
Comment (chibenwa) [inline server/protocols/jmap-draft-integration-testing/jmap-draft-integration-testing-common/src/test/java/org/apache/james/jmap/draft/methods/integration/SpamAssassinContract.java:60]: Please do not throw away this test, relocate it into `third-party/spamassassin` (only keep the memory ones)

**Review by chibenwa — COMMENTED** [review]
You forgot `SpamAssassinHandler` ?

Also, `server/apps/distributed-app/docs/modules/ROOT/partials/SpamAssassin.adoc` and `server/apps/distributed-app/docs/modules/ROOT/pages/configure/spam.adoc` needs to be modified to reflect your changes.

Don't forget `upgrade-instructions.md` too...

File: third-party/spamassassin/pom.xml [hunk 2]
```diff
@@ -30,6 +30,10 @@
     <name>Apache James :: Third Party :: SpamAssassin</name>
     <description>SpamAssassin Java client and testing utilities</description>
 
+    <properties>
+        <james.baseVersion>${project.version}</james.baseVersion>
```
Comment (Arsnael) [inline third-party/spamassassin/pom.xml:34]: What for? You already have `project.version` available

File: third-party/spamassassin/pom.xml [hunk 3]
```diff
@@ -84,6 +88,17 @@
...
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-common</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-guice-configuration</artifactId>
+            <version>${james.baseVersion}</version>
```
Comment (Arsnael) [inline third-party/spamassassin/pom.xml:100]: Is this really necessary? Logically you should not have to redeclare versions here? Idem for other concerned deps in this pom

Comment (vttranlina) [reply third-party/spamassassin/pom.xml:100]: I copied from the `examples` module. But yes, It's not necessary

**Comment by chibenwa** [comment]
You have a conflict.

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-1108/7/testReport/

Bunch of failure on some spam matcher... is it related?

**Comment by chibenwa** [comment]
Yes, likely!

**Pull request merged**